runtime/shared_common/shrinit.cpp

```diff
@@ -4251,6 +4329,16 @@ j9shr_shutdown(J9JavaVM *vm)
 		if (utfHashTable) {
 			hashTableFree(utfHashTable);
 		}
+		if (NULL != romToRamHashTable) {
```

Comment:
Talked to @theresa-m, it is possible that the hashtable is freed by another thread after line 4286, so this check and the following `if (romToRamHashTable) {` won't work as it is checking the local variable `romToRamHashTable` whose value might become stale already.